evalml/tests/component_tests/test_arima_regressor.py

```diff
@@ -472,3 +474,58 @@ def test_arima_regressor_can_forecast_arbitrary_dates(use_covariates, ts_data):
     assert (
         arima.predict(X_test).tail(5).tolist() == arima.predict(X_test_last_5).tolist()
     )
+
+
+def test_arima_regressor_nullable_handling(ts_data):
+    # ARIMARegressor currently has handling in place, so no error will occur from nullable types
+    X = pd.DataFrame()
+    X["nums"] = pd.Series([i for i in range(100)], dtype="Int64")
+    X.index = pd.date_range("1/1/21", periods=100)
+    X.ww.init(logical_types={"nums": "IntegerNullable"})
+
+    y = pd.Series([i for i in range(100)], dtype="float64")
+    y.index = pd.date_range("1/1/21", periods=100)
+
+    X_train = X.ww.iloc[:80, :]
+    X_test = X.ww.iloc[80:, :]
+
+    y_train = y[:80]
+
+    arima_params = {
+        "trend": None,
+        "start_p": 2,
+        "d": 0,
+        "start_q": 2,
+        "max_p": 5,
+        "max_d": 2,
+        "max_q": 5,
+        "seasonal": True,
+        "maxiter": 10,
+        "n_jobs": -1,
+    }
+
+    # SKTime's AutoARIMA regressor cannot handle IntegerNullable type
+    sk_arima = SKArima(**arima_params)
+    fh_ = ForecastingHorizon(
+        [i + 1 for i in range(len(X_test))],
+        is_relative=True,
+    )
+
+    with pytest.raises(
+        ValueError,
+        match="Could not successfully fit a viable ARIMA model to input data.",
+    ):
+        # Will stop raising this error when sktime adds support for IntegerNullable - https://github.com/sktime/sktime/issues/4236
+        # TODO: Remove nullable type handling https://github.com/alteryx/evalml/issues/4016
+        sk_arima.fit(y=y_train, X=X_train)
+
+    # But EvalML's ARIMARegressor has a call to downcast_int_nullable_to_double to handle IntegerNullable
+    # so we don't see any errors currently
+    evalml_arima = ARIMARegressor(**arima_params)
+    evalml_arima.fit(X_train, y_train)
+
+    # Confirm that the handle nullable types method fixes the error for AutoARIMA
+    X_train_d, y_train_d = evalml_arima._handle_nullable_types(X_train, y_train)
```

Comment:
Once we integrate the new handling into automl search, this call to `_handle_nullable_types` will be inside fit and predict, so we'll remove this. Keeping the error checking for sk_arima will help us know when sktime adds support for integer nullable on arima.